geopandas/vectorized.pyx

```diff
@@ -1093,6 +1103,23 @@ cdef geom_type(np.ndarray[np.uintp_t, ndim=1, cast=True] geoms):
     return out
 
 
+cdef cy_unary_union(np.ndarray[np.uintp_t, ndim=1, cast=True] geoms):
+    cdef GEOSContextHandle_t handle
+    cdef GEOSGeometry *collection
+    cdef GEOSGeometry *out
+    cdef size_t n = geoms.size
+
+    handle = get_geos_context_handle()
+
+    with nogil:
+        collection = GEOSGeom_createCollection_r(handle, GEOS_MULTIPOLYGON,
+                                                 <GEOSGeometry **> geoms.data,
+                                                 n)
+        out = GEOSUnaryUnion_r(handle, collection)
+
+    return geom_factory(<np.uintp_t> out)
```

Comment:
Do we need to 'destroy' the collection inside this function? (as it not further used) 
Or not because the geoms are still inside the original array? 

I have to say I don't fuly understand the comment in geos_c.h:

```
 /*
  499  * Second argument is an array of GEOSGeometry* objects.
  500  * The caller remains owner of the array, but pointed-to
  501  * objects become ownership of the returned GEOSGeometry.
  502  */
```